File: pyg_parent/pyg_interface/src/main/java/com/itheima/pyg/service/itemcat/ItemCatService.java
package com.itheima.pyg.service.itemcat;

import com.itheima.pyg.pojo.item.ItemCat;

import java.util.List;

public interface ItemCatService {
    /**
     * 向数据库添加分类,用于excel数据导入数据库
     * @param itemCat
     */
    void save(ItemCat itemCat);
    /**
     * 根据父ID查询分类列表
     * @param parentId
     * @return
     */
    List<ItemCat> findByParentId(long parentId);

    /**
     * 查询商品分类实体
     * @param id
     * @return
     */
    ItemCat findOne(long id);

    /**
     * 查询全部商品分类
     * @return
     */
    List<ItemCat> findAll();
}

File: pyg_parent/pyg_web_shop/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../itemCat/findAll.action');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../itemCat/findPage.action?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../itemCat/findOne.action?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../itemCat/add.action',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../itemCat/update.action',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../itemCat/delete.action?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../itemCat/search.action?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    	
	
	this.findByParentId = function(parentId){
		return $http.get("../itemCat/findByParentId.action?parentId="+parentId);
	}
});
